astropy/modeling/core.py

```diff
@@ -1224,8 +1365,115 @@ def _param_sets(self, raw=False):
             psets[:] = values
             return psets
 
+        # TODO: Returning an array from this method may be entirely pointless
+        # for internal use--perhaps only the external param_sets method should
+        # return an array (and just for backwards compat--I would prefer to
+        # maybe deprecate that method)
+
         return np.array(values)
 
+    def _prepare_input_units(self, inputs):
+        for input_ in inputs:
```

Comment:
Yeah, I like that.  I think originally there was more going on this loop, but after I made some simplifications it was nothing more than this.